core/blockchain.go

```diff
@@ -800,8 +800,12 @@ func (bc *BlockChain) Stop() {
 
 		for _, offset := range []uint64{0, 1, TriesInMemory - 1} {
 			if number := bc.CurrentBlock().NumberU64(); number > offset {
-				recent := bc.GetBlockByNumber(number - offset)
-
+				num := number - offset
+				recent := bc.GetBlockByNumber(num)
+				if recent == nil {
+					log.Error("Failed to get block", num)
+					continue
+				}
```

Comment:
We'd rather revert this -- is this path triggered by the tests? 
Also, the log format is wrong, would need to be `log.Error("Failed to get block", "num", num)`

Reply:
As far as I could tell, the tests didn't trigger this path, so I have reverted this change. If this happens, it indicates that something is _seriously_ wrong, either our blockchain is corrupt, or the database is closed.